src/ucp/core/ucp_mm.inl

```diff
@@ -73,13 +74,19 @@ ucp_memh_put(ucp_context_h context, ucp_mem_h memh, int invalidate)
     }
 
     UCP_THREAD_CS_ENTER(&context->mt_lock);
-    if (invalidate) {
-        ucs_rcache_region_invalidate(context->rcache, &memh->super,
-                (ucs_rcache_invalidate_comp_func_t)ucs_empty_function, NULL);
+    if (cb != NULL) {
+        memh->flags &= UCP_MEMH_FLAG_INVALIDATE;
```

Comment:
looks weird
did you want to remove this flag or add it? I would not expect we need to remove it